src/client/Microsoft.Identity.Client/Internal/Requests/SilentRequestHelper.cs

```diff
@@ -101,14 +105,29 @@ internal static bool NeedsRefresh(MsalAccessTokenCacheItem oldAccessToken, out D
                     {
                         logger.ErrorPiiWithPrefix(ex, logMsg);
                     }
+
+                    error = ex.ErrorCode;
                 }
                 catch (OperationCanceledException ex)
                 {
                     logger.WarningPiiWithPrefix(ex, ProactiveRefreshCancellationError);
+                    error = ex.GetType().Name;
                 }
                 catch (Exception ex)
                 {
                     logger.ErrorPiiWithPrefix(ex, ProactiveRefreshGeneralError);
+                    error = ex.GetType().Name;
+                }
+                finally
```

Comment:
I don't think a finally block is needed. Error var is only set in catch clauses. Finally executes for non-error cases for nothing.

Why not have a private method `LogOtelError(error, apidId..)` and call it in every `catch` clause?